pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2328,6 +2330,61 @@ public void readEntryComplete(Entry entry, Object ctx) {
         }
     }
 
+    protected void internalGetMessageIdByTimestamp(AsyncResponse asyncResponse, long timestamp, boolean authoritative) {
```

Comment:
It's better to return CompletableFuture<MessageIdImpl> directly, it will make the code easier to read. you can see `PersistentTopics.setMaxSubscriptionsPerTopic`